app/controllers/BankDetailsController.scala

```diff
@@ -99,31 +102,47 @@ class BankDetailsController @Inject() (
   }
 
   val enterBankDetailsSubmit: Action[AnyContent] = as.eligibleJourneyAction.async { implicit request =>
-    BankDetailsForm.form
-      .bindFromRequest()
-      .fold(
-        formWithErrors =>
-          Future.successful(Ok(views.enterBankDetailsPage(formWithErrors, BankDetailsController.chooseTypeOfAccountUrl))),
-        (bankDetailsForm: BankDetailsForm) => {
-
-          val directDebitDetails: DirectDebitDetails = DirectDebitDetails(
-            BankDetails(
-              name          = bankDetailsForm.name,
-              sortCode      = bankDetailsForm.sortCode,
-              accountNumber = bankDetailsForm.accountNumber
-            ),
-            bankDetailsForm.isSoleSignatory.asBoolean
-          )
-
-          journeyService.updateDirectDebitDetails(request.journeyId, directDebitDetails)
-            .map { _ =>
-              bankDetailsForm.isSoleSignatory match {
-                case IsSoleSignatoryFormValue.Yes => Redirect(routes.BankDetailsController.checkBankDetails)
-                case IsSoleSignatoryFormValue.No  => Redirect(routes.BankDetailsController.cannotSetupDirectDebitOnlinePage)
-              }
+    val validForm = BankDetailsForm.form.bindFromRequest()
```

Comment:
I think this should be called something other than validForm, as we fold on it to check for invalid form, if that makes sense

Reply:
if you're only using this `val` once then you may as well just not have it. If you need access to it below, you could probably just use what you've called `bankDetailsForm` below

Reply:
oh yeah 🙂 agreed